src/main/java/org/red5/server/net/rtmp/RTMPConnection.java

```diff
@@ -122,19 +125,19 @@
 	 * 
 	 * @see org.red5.server.net.rtmp.Channel
 	 */
-	private transient ConcurrentMap<Integer, Channel> channels = new ConcurrentHashMap<Integer, Channel>(3, 0.9f, 1);
+	private transient ConcurrentMap<Integer, Channel> channels = new ConcurrentHashMap<Integer, Channel>(8, 0.9f, 4);
 
 	/**
 	 * Client streams
 	 * 
 	 * @see org.red5.server.api.stream.IClientStream
 	 */
-	private transient ConcurrentMap<Integer, IClientStream> streams = new ConcurrentHashMap<Integer, IClientStream>(1, 0.9f, 1);
+	private transient ConcurrentMap<Integer, IClientStream> streams = new ConcurrentHashMap<Integer, IClientStream>(4, 0.9f, 2);
 
 	/**
 	 * Reserved stream ids. Stream id's directly relate to individual NetStream instances.
 	 */
-	private volatile BitSet reservedStreams = new BitSet();
+	private Set<Integer> reservedStreams = Collections.newSetFromMap(new ConcurrentHashMap<Integer, Boolean>(4, 0.9f, 2));
 

```

Comment:
I relied on that topic http://stackoverflow.com/questions/6720396/different-types-of-thread-safe-sets-in-java. I was choosing between ConcurrentSkipListSet and Collections.newSetFromMap() and second implementation showed better execution time for our case.

I checked documentation and source code (on grepcode.com) for this method:
>Returns a set backed by the specified map. The resulting set displays the same ordering, concurrency, and performance characteristics as the backing map. In essence, this factory method provides a Set implementation corresponding to any Map implementation. 

As I understand this set is just wrapper around specified map so it can be used in concurrent environment.